UtopiaCity/Controllers/CitizenAccount/CitizensAccountController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using UtopiaCity.Data;
+using UtopiaCity.Models.CitizenAccount;
+
+namespace UtopiaCity.Controllers.CitizenAccount
+{
+    [Authorize]
+    public class CitizensAccountController : Controller
+    {
+        private readonly ApplicationDbContext _dbContext;
+        private UserManager<AppUser> _userManager;
+        private readonly SignInManager<AppUser> _signInManager;
+        public CitizensAccountController(ApplicationDbContext dbContext, UserManager<AppUser> userManager, SignInManager<AppUser> signInManager)
+        {
+            _dbContext = dbContext;
+            _userManager = userManager;
+            _signInManager = signInManager;
+        }
+        public IActionResult ShowUserCabinet()
+        {
+            return View();
+        }
+
+        public async Task<IActionResult> IndexAsync()
+        {
+            AppUser user = await _userManager.GetUserAsync(User);
+            return View(user);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> EditAsync(string id)
```

Comment:
can't understand for what you're using id in this method, and for what you' ve created service class"CitizenAccount" if you're not using it , and how do you understand which model of User will be edited ?

Reply:
Hi!
1. in this method, I use a parameter to check for the occupancy of this parameter, just as we did in the lesson when working with the emergencyreport controller. That is, if we pass the parameter in the form https://localhost:44375/CitizensAccount/Edit/83b43606-a779-474f-8e83-9346825739d1 everything will be ok, the method will work. And if we pass https://localhost:44375/CitizensAccount/Edit the method does not understand which representative of the User class to work with. But in this case, the entire controller has the Authorize attribute, and in principle, we will not get into this method without authorization. So perhaps this parameter is unnecessary.

2. I created this class for other developers, I saw use-case where other developer need find User by Id for their use-case. And also, I think if I will must append add functionality it will be easier.I don't use functionality of class "CitizensAccountService" because class "AppUser" extended from "IdentityUser" and for this class microsoft created perfect methods and I use them.

3. How I wrote before If U want use methods of class "CitizensAccountController" U must be authorized. And all controllers has "User" property. And this User is exactly the one that is needed.

Reply:
Ibraga96, 
While I was answering you,I think about parameter ID in Edit method, And I think it redundant.
Thanks:)

Reply:
no problems, if i understood correctly you have extended, templated class User , and use this class, and also you'll have your own authorization form . correct me if i wrong

Reply:
Yes everything right. But only I don't have specific authorization form, it is standart form by microsoft

Reply:
> Hi!
> 
> 1. in this method, I use a parameter to check for the occupancy of this parameter, just as we did in the lesson when working with the emergencyreport controller. That is, if we pass the parameter in the form https://localhost:44375/CitizensAccount/Edit/83b43606-a779-474f-8e83-9346825739d1 everything will be ok, the method will work. And if we pass https://localhost:44375/CitizensAccount/Edit the method does not understand which representative of the User class to work with. But in this case, the entire controller has the Authorize attribute, and in principle, we will not get into this method without authorization. So perhaps this parameter is unnecessary.
> 2. I created this class for other developers, I saw use-case where other developer need find User by Id for their use-case. And also, I think if I will must append add functionality it will be easier.I don't use functionality of class "CitizensAccountService" because class "AppUser" extended from "IdentityUser" and for this class microsoft created perfect methods and I use them.
> 3. How I wrote before If U want use methods of class "CitizensAccountController" U must be authorized. And all controllers has "User" property. And this User is exactly the one that is needed.

we can find a current user.
var currentUser = User.Identity.Name;
And name can store that user id.

Reply:
we can find a current user.
var currentUser = User.Identity.Name;
And name can store that user id.

Reply:
we can assign id to Claims. and claims pass to token. Tokens configured in StartUp.